Mathlib/Data/List/SplitLengths.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Order.MinMax
+import Mathlib.Algebra.Group.Nat
+
+/-!
+# Splitting a list to chunks of specified lengths
+
+This file defines splitting a list to chunks of given lengths, and some proofs about that.
+-/
+
+variable {α : Type*} (l : List α) (sz : List ℕ)
+
+namespace List
+
+/--
+Split a list to chunks of given lengths.
+-/
+def splitLengths : List ℕ → List α → List (List α)
+  | [], _ => []
+  | n::ns, x =>
+    let (x0, x1) := x.splitAt n
+    x0 :: ns.splitLengths x1
+
+@[simp]
+theorem length_splitLengths :
+    (sz.splitLengths l).length = sz.length := by
+  induction sz generalizing l
+  · simp [splitLengths]
+  · simp [splitLengths, ‹∀ (l : List α), _›]
```

Comment:
Out of curiosity, what's `‹∀ (l : List α), _›` accomplishing here?

Reply:
It's adding the induction hypothesis to the `simp`

Reply:
Would it be cleaner to use the `induction ... with` syntax to name the inductive hypothesis explicitly?